sklearn/metrics/tests/test_score_objects.py

```diff
@@ -630,11 +632,14 @@ def test_classification_scorer_sample_weight():
         else:
             target = y_test
         try:
+            scorer = scorer.set_score_request(sample_weight=True)
             weighted = scorer(
                 estimator[name], X_test, target, sample_weight=sample_weight
             )
             ignored = scorer(estimator[name], X_test[10:], target[10:])
             unweighted = scorer(estimator[name], X_test, target)
+            # this should not raise. sample_weight should be ignored if None.
+            _ = scorer(estimator[name], X_test[:10], target[:10], sample_weight=None)
```

Comment:
Are we going to change this behavior in 1.3? If so, can we leave a TODO here?

Reply:
no this one is expected behavior.